File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Python_Solutions/Problem5.py
import math

def newton_raphson(x0, tolerance=0.0001, max_iterations=10):
    x = x0
    print(f"Newton-Raphson Method for f(x) = e^x - 3x = 0")
    print(f"Initial guess: x₀ = {x0}")
    print("\nIterations:")
    
    for i in range(max_iterations):
        fx = math.exp(x) - 3*x    # f(x) = e^x - 3x
        fpx = math.exp(x) - 3     # f'(x) = e^x - 3
        
        x_next = x - fx / fpx
        
        print(f"Iteration {i+1}: x = {x:.6f}, f(x) = {fx:.6f}")
        
        if abs(x_next - x) < tolerance:
            print(f"\nRoot found: {x_next:.6f}")
            break
            
        x = x_next

if __name__ == "__main__":
    newton_raphson(1.0)  # Initial guess 

File: Numerical_Methods/Numerical Solutions of Algebraic Equations/Newton-Raphson Method/Java_Solutions/Problem5.java
public class Problem5 {
    public static void main(String[] args) {
        double x0 = 1.0;  // Initial guess
        double tolerance = 0.0001;
        int maxIterations = 10;
        
        System.out.println("Newton-Raphson Method for f(x) = e^x - 3x = 0");
        System.out.println("Initial guess: x₀ = " + x0);
        System.out.println("\nIterations:");
        
        double x = x0;
        for (int i = 0; i < maxIterations; i++) {
            double fx = Math.exp(x) - 3 * x;    // f(x) = e^x - 3x
            double fpx = Math.exp(x) - 3;       // f'(x) = e^x - 3
            
            double xNext = x - fx / fpx;
            
            System.out.printf("Iteration %d: x = %.6f, f(x) = %.6f\n", 
                            i + 1, x, fx);
            
            if (Math.abs(xNext - x) < tolerance) {
                System.out.printf("\nRoot found: %.6f\n", xNext);
                break;
            }
            
            x = xNext;
        }
    }
} 